### ISSUE TITLE
PrecompiledAssemblyException: Multiple precompiled assemblies with the same name Newtonsoft.Json.dll 

### ISSUE THREAD

**Issue by AkshathaSP** [BrianPeek/AzureSDKs-Unity#4]
Imported Azure SDK in Unity. Build fails with error:

PrecompiledAssemblyException: Multiple precompiled assemblies with the same name Newtonsoft.Json.dll included for the current platform. Only one assembly with the same name is allowed per platform. Assembly path: {0}
UnityEditor.Scripting.ScriptCompilation.EditorBuildRules.CreateTargetAssemblies (System.Collections.Generic.IEnumerable`1[T] customScriptAssemblies, System.Collections.Generic.IEnumerable`1[T] precompiledAssemblies) (at C:/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/EditorBuildRules.cs:249)
UnityEditor.Scripting.ScriptCompilation.EditorCompilation.UpdateCustomTargetAssemblies () (at C:/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/EditorCompilation.cs:619)
UnityEditor.Scripting.ScriptCompilation.EditorCompilation.SetAllCustomScriptAssemblyJsons (System.String[] paths) (at C:/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/EditorCompilation.cs:728)
UnityEditor.Scripting.ScriptCompilation.EditorCompilationInterface.SetAllCustomScriptAssemblyJsons (System.String[] allAssemblyJsons) (at C:/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/EditorCompilationInterface.cs:217)

**Comment by AkshathaSP** [comment]
Unity version used: 2018.3.2f1

**Comment by BrianPeek** [comment]
Hello, looks like you might be trying to use multiple Azure SDKs at the same time, or you already have something dependent on Newtonsoft.Json in your project?  If so, try removing the duplicate copy of Newtonsoft.Json, which may fix the issue.  However, you may run into an issue where multiple versions of Newtonsoft.Json are dependent, in which case I unfortunately don't yet have a workaround for.

**Comment by AkshathaSP** [comment]
Hi Brian,
I created a new unity project and added these Assests: HololensToolkit , AzureStorageSDK for unity.
The build was successful when I added HololensToolkit. It started failing after importing AzureStorage package. So i removed HololensToolkit and the issue still exists. Is there any settings where we can configue that Precompiled asssemblies of the same name should keep only single reference ?
